_posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md

```diff
@@ -0,0 +1,268 @@
+   VM-exit controls. Address-range monitoring is cleared.
+5. MSRs may be loaded from the VM-exit MSR-load area.
+
+As you can see, there is no explicit error checking performed - it doesn't make
+sense to return back to VM anyway in this case. An error can happen when for any
+reason VMCS gets corrupted (memory range containing VMCS was not properly
+protected from VMs, different core in VMM mode or another hardware through DMA
+corrupted VMCS, physical memory error). It isn't possible to return to a non-VMX
+state because it is not saved anywhere - host-VM is still only a VM, it is
+indistinguishable from any other VM at this time. CPU is put into a shutdown
+state and only RESET signal can bring it back. All this is done to prevent any
+chance of privilege elevation - safety is considered more important than
+continued operation in this case.
+
+## VMCS
+
+VM control structure. It is pointed to by VMCS pointer - one per logical
+processor, which is the main reason why VMs are limited to one core from the
+developer's point of view. There is always one VMCS per VM, even on symmetric
+implementations, because some of its fields describe CPU state at the time of
+the transition between VM and VMM so they cannot be shared by multiple cores.
+
+The exact layout of this structure, as well as its size,  is implementation
+specific. For this reason, as well as because it can be internally cached by
+CPU for better performance, fields of VMCS should not be accessed directly.
+Special instructions defined by VMX should be used instead (VMCLEAR, VMREAD and
+VMWRITE).
+
+The VMCS data is organized into six logical groups:
+
+* **Guest-state area** - processor state is saved into the guest-state area on
+  VM exits and loaded from there on VM entries.
+* **Host-state area** - processor state is loaded from the host-state area on
+  VM exits. It is usually saved only once when creating VMCS.
+* **VM-execution control fields** - these fields control processor behaviour in
+  VMX non-root operation. They determine in part the causes of VM exits.
+* **VM-exit control fields** - these fields control VM exits.
+* **VM-entry control fields** - these fields control VM entries.
+* **VM-exit information fields** - these fields receive information on VM exits
+  and describe the cause and the nature of VM exits. On some processors, these
+  fields are read-only.
+
+The VM-execution control fields, the VM-exit control fields, and the VM-entry
+control fields are sometimes referred to collectively as VMX controls.
+
+An overview of fields available in VMCS is available [here](https://github.com/LordNoteworthy/cpu-internals/raw/master/VMCS.pdf).
+Note that not all of those fields are available on all processors, also new ones
+can be added in the future.
+
+VMCS can be in one of six launch states, depending mostly on the last
+VMCS-related instruction called:
+
+![VMCS states](https://3mdeb.com/wp-content/uploads/2019/04/vmcs_states.png)
```

Comment:
link is dead

Reply:
is it proper path now?

Reply:
@krystian-hebel Looks good. You can always check if the post is rendering correctly by setting local server - step by step instruction is located in `README.md` file